core/trino-spi/src/main/java/io/trino/spi/block/FixedBinaryBlock.java

```diff
@@ -0,0 +1,270 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.block;
+
+import io.airlift.slice.Slice;
+import io.airlift.slice.Slices;
+import io.trino.spi.block.FixedBinaryBlockEncoding.FixedBinaryBlockOutput;
+import org.openjdk.jol.info.ClassLayout;
+
+import javax.annotation.Nullable;
+
+import java.util.Optional;
+import java.util.function.BiConsumer;
+
+import static io.airlift.slice.SizeOf.sizeOf;
+import static io.trino.spi.block.BlockUtil.checkArrayRange;
+import static io.trino.spi.block.BlockUtil.checkValidRegion;
+import static io.trino.spi.block.BlockUtil.compactArray;
+import static io.trino.spi.block.BlockUtil.countUsedPositions;
+
+public class FixedBinaryBlock
+        implements Block, FixedBinaryBlockOutput
+{
```

Comment:
I think we can make it super simple, see `io.trino.plugin.geospatial.BingTileType`